src/yang.erl

```diff
@@ -1626,23 +1627,31 @@ get_identity(RawRef, Pos, Identites, M, Ctx) ->
 %%     get_definition(RawRef, Pos, Extensions, PrefixMap, Ctx,
 %%                    fun(M) -> M#module.extensions end, 'extension').
 
-get_definition(RawRef, Pos, DefinitionMap, M, Ctx0, GetMapF, Kind) ->
+get_definition(RawRef, Pos, DefinitionMap, M, Status, Ctx0, GetMapF, Kind) ->
     case resolve_raw_idref(RawRef, Pos, M, Ctx0) of
         {self, RefName, Ctx1} ->
             %% reference to local definition
-            get_definition2(RefName, Pos, DefinitionMap, Ctx1, Kind);
+            get_definition2(RefName, Pos, DefinitionMap, Status, Ctx1, Kind);
         {{imported, RefM}, RefName, Ctx1} ->
             get_definition2({RefM#module.name, RefName},
-                            Pos, GetMapF(RefM), Ctx1, Kind);
+                            Pos, GetMapF(RefM), undefined, Ctx1, Kind);
         {undefined, _RefName, Ctx1} ->
             {undefined, Ctx1}
     end.
 
-get_definition2(DefinitionYangIdentifier, Pos, DefinitionMap, Ctx0, Kind) ->
+get_definition2(DefinitionYangIdentifier, Pos, DefinitionMap, Status, Ctx0, Kind) ->
     DefinitionName = local_name(DefinitionYangIdentifier),
     case map_lookup(DefinitionName, DefinitionMap) of
         {value, Definition} ->
-            {Definition, Ctx0};
+            if Kind == 'feature' andalso Status /= undefined ->
```

Comment:
This doesn't look quite right.  I think it would be better to not do this status check here, but instead do it in chk_if_feature.